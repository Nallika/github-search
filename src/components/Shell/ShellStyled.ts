import styled from 'styled-components';

const Wrapper = styled.div`
  padding: 20px;
  height: 100%;
`;

export { Wrapper }
