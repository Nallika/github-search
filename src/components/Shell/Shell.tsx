import React from 'react';

import { SearchProvider } from '../SearchContext';
import ResultsList from '../ResultsList';
import SearchBar from '../SearchBar';
import { Wrapper } from './ShellStyled';

/**
 *
 * @constructor
 */
const Shell = () => {
  return (
    <SearchProvider>
      <Wrapper>
        <SearchBar />
        <ResultsList />
      </Wrapper>
    </SearchProvider>
  );
}

export default Shell;
