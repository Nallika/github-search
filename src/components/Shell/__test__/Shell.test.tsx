import React, {ReactNode} from 'react';
import { render } from '@testing-library/react';
import Shell from '../Shell';

jest.mock('../../SearchContext', () => ({
  SearchProvider: ({ children }: { children: ReactNode }) => <div>{children}</div>,
}));
jest.mock('../../ResultsList', () => () => <div>ResultsList</div>);
jest.mock('../../SearchBar', () => () => <div>SearchBar</div>);

test('`<Shell />` Should render component', () => {
  const { container } = render(<Shell />);

  expect(container).toMatchSnapshot();
});
