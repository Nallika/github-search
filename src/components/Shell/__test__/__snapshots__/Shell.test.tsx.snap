// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`<Shell />\` Should render component 1`] = `
<div>
  <div>
    <div
      class="sc-bdfCDU dvMobY"
    >
      <div>
        SearchBar
      </div>
      <div>
        ResultsList
      </div>
    </div>
  </div>
</div>
`;
